### PR TITLE
 `Development`: Provide web-based problem statement and feedback view to native clients.

### PR DESCRIPTION
**Pull request by TimOrtel**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

The problem statement and the feedback view are currently to complex to move them to the native clients with a native implementation. Therefore, we want to embed them using web views.
### Description

We refactor the problem statement to be able to exist on its own. Furthermore, we add a new component that can show the feedback component based on query args.

### Steps for Testing
0. Deploy to TestServer2 (since the id's below are only valid for testServer 2, otherwise you can use another test server as well but research the exerciseID , participationID and resultId yourself)
1. Login with Artemis_admin account
2. Go to https://artemis-test2.artemis.in.tum.de/courses/567/exercises/1513/problem-statement/3031
3. Verify that you can see the problem statement end it looks fine (only works for light mode, dark mode is handled in native clients (background is transparent)
4. Go to https://artemis-test2.artemis.in.tum.de/courses/567/exercises/1513/feedback/3031/3931/true
5. Verify that you can see the feedback to your result (only works for light mode, dark mode is handled in native clients (background is transparent)
6. Also verify that https://artemis-test2.artemis.in.tum.de/courses/567/exercises/1513/ still looks like it should

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| problem-statement.component.ts | 82% | 96% |
| standalone-feedback.component.ts | 77% | 98% |

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/feedback/feedback.component.html [hunk 1]
```diff
@@ -130,5 +130,5 @@ <h4 jhiTranslate="artemisApp.result.noResultDetails">No result details available
     </ng-template>
 </div>
 <div class="modal-footer">
-    <button id="feedback-close" type="button" class="btn btn-outline-dark" (click)="activeModal.close('Close click')">Close</button>
+    <button *ngIf="activeModal != null" id="feedback-close" type="button" class="btn btn-outline-dark" (click)="activeModal.close('Close click')">Close</button>
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/feedback/feedback.component.html:133]:
```suggestion
    <button *ngIf="activeModal" id="feedback-close" type="button" class="btn btn-outline-dark" (click)="activeModal.close('Close click')">Close</button>
```

File: src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<jhi-result-detail
+    *ngIf="exercise != null && result != null && exerciseType != null"
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.html:2]:
```suggestion
    *ngIf="exercise && result && exerciseType"
```

File: src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit, Optional, ViewChild } from '@angular/core';
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts:1]: Unused import

File: src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    constructor(public route: ActivatedRoute, private exerciseService: ExerciseService, @Optional() private exerciseCacheService: ExerciseCacheService) {}
+
+    ngOnInit(): void {
+        this.route.params.subscribe((params) => {
+            const exerciseId = parseInt(params['exerciseId'], 10);
+            const participationId = parseInt(params['participationId'], 10);
+            const resultId = parseInt(params['resultId'], 10);
+
+            this.isTemplateStatusMissing = params['isTemplateStatusMissing'] == 'true';
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts:34]:
```suggestion
            this.isTemplateStatusMissing = params['isTemplateStatusMissing'] === 'true';
```

File: src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+            this.exerciseService.getExerciseDetails(exerciseId).subscribe((exerciseResponse: HttpResponse<Exercise>) => {
+                this.exercise = exerciseResponse.body!;
+                const participation = this.exercise?.studentParticipations?.find((participation) => participation.id == participationId);
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts:38]:
```suggestion
                const participation = this.exercise?.studentParticipations?.find((participation) => participation.id === participationId);
```

File: src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                if (participation != null) {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts:39]:
```suggestion
                if (participation) {
```

File: src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                    participation.exercise = this.exercise;
+                }
+
+                const relevantResult = participation?.results?.find((result) => result.id == resultId);
+                if (relevantResult != null) {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts:44]:
```suggestion
                if (relevantResult) {
```

File: src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+            (this.exerciseCacheService ?? this.exerciseService).getLatestDueDate(exerciseId).subscribe((latestDueDate) => {
+                this.latestDueDate = latestDueDate;
+                this.setup();
+            });
+        });
+    }
+
+    private setup() {
+        if (this.exercise != null && this.result != null) {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/feedback/standalone-feedback/standalone-feedback.component.ts:61]:
```suggestion
        if (this.exercise && this.result) {
```

File: src/main/webapp/app/shared/layouts/main/main.component.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import { Component, OnInit } from '@angular/core';
-import { ActivatedRouteSnapshot, NavigationEnd, NavigationError, Router } from '@angular/router';
+import { ActivatedRoute, ActivatedRouteSnapshot, NavigationEnd, NavigationError, Router } from '@angular/router';
```
Comment (JohannesStoehr) [inline src/main/webapp/app/shared/layouts/main/main.component.ts:2]: Unused import

**Review by JohannesStoehr — DISMISSED** [review]

**Review by mertyldrr — APPROVED** [review]
tested in the testing session. problem statement and feedback texts were identical

**Review by frederik-kt — DISMISSED** [review]
Tested on TS2. Problem statement and feedback view were accessible without header, footer, and background. Going back to Artemis, I clicked around the UI and everything still worked as expected. Header and footer were still visible.

**Review by terlan98 — DISMISSED** [review]
Tested on TS2 during a testing session. Works as expected

**Comment by sven0311** [comment]
Steps for testing:
1. Login with Artemis_admin account
2. Go to https://artemis-test2.artemis.in.tum.de/courses/567/exercises/1513/problem-statement/3031
3. Verify that you can see the problem statement end it looks fine (only works for light mode, dark mode is handled in native clients (background is transparent)
4. Go to https://artemis-test2.artemis.in.tum.de/courses/567/exercises/1513/feedback/3031/3931/true
5. Verify that you can see the feedback to your result

**Review by sven0311 — APPROVED** [review]
Tested it during the testing session on ts2

**Review by YeetTheFirst21 — APPROVED** [review]
Seems to be working as expected, even checked on mobile as that seems like the best use case for it.

Screenshot 2023-04-02 173701

Screenshot_20230402-173651_Chrome

File: src/test/javascript/spec/component/shared/feedback/standalone-feedback.ocmponents.spec.ts
Comment (terlan98) [inline src/test/javascript/spec/component/shared/feedback/standalone-feedback.ocmponents.spec.ts:1]: I think this file should be renamed to 'standalone-feedback.component.spec.ts'

**Review by terlan98 — APPROVED** [review]
Retested on TS2 and reapproved. Works fine 👍

**Comment by sven0311** [comment]
Reopened in a [new PR](https://github.com/ls1intum/Artemis/pull/6427) since CI does not work for forks

**Pull request closed without merging**